src/components/data/row/SwapProviderRow.tsx

```diff
@@ -0,0 +1,52 @@
+import { EdgeSwapQuote } from 'edge-core-js'
+import React from 'react'
+import FastImage from 'react-native-fast-image'
+
+import { useCryptoText } from '../../../hooks/useCryptoText'
+import { lstrings } from '../../../locales/strings'
+import { getSwapPluginIconUri } from '../../../util/CdnUris'
+import { cacheStyles, Theme, useTheme } from '../../services/ThemeContext'
+import { EdgeText } from '../../themed/EdgeText'
+import { IconDataRow } from './IconDataRow'
+
+export interface Props {
+  quote: EdgeSwapQuote
+}
+
+export const SwapProviderRow = (props: Props) => {
+  const { quote } = props
+  const { request, toNativeAmount, fromNativeAmount } = quote
+  const { quoteFor } = request
+  const { fromWallet, fromTokenId, toWallet, toTokenId } = request
+  const theme = useTheme()
+  const styles = getStyles(theme)
+
+  const isToQuote = quoteFor === 'to'
+  const wallet = isToQuote ? fromWallet : toWallet
+  const tokenId = isToQuote ? fromTokenId : toTokenId
+  const nativeAmount = isToQuote ? fromNativeAmount : toNativeAmount
+
+  const toAmount = useCryptoText({ wallet, tokenId, nativeAmount })
```

Comment:
Misnomer. Might be a 'from' or 'to'. Should be `exchangeAmount`

Reply:
Made it a bit more literal, 'exchange' already has multiple meanings in our codebase

Reply:
I'll accept the update, but `exchangeAmount` has a very specific meaning. I might agree that it's not the best term but we do use it consistently.